src/FilterImages.js

```diff
@@ -0,0 +1,47 @@
+import Component from './Component.js';
+
+class FilterCreatures extends Component {
+    
+    onRender(select) {
+        const onFilter = this.props.onFilter;
+        select.addEventListener('input', () => {
+            onFilter(select.value);
+        });
+    }
+
+    renderHTML() {
+        const hornedCreatures = this.props.hornedCreatures;
+        const types = getUniqueTypes(hornedCreatures);
+        const optionsHTML = renderOptionsHTML(types);
+
+        return /*html*/`
+        <select id="drop-down">
+            <option value="all">ALL</option>
+            ${optionsHTML}
+        </select>
+        `;
+    }
+}
+
+
+function getUniqueTypes(hornedCreatures) {
+    const hornsArray = [];
+    hornedCreatures.forEach(creatures => {
```

Comment:
when using forEach, you should use the singular to represent each item in the array for clarity's sake (ie "creatures" should be "creature" here).